app/Database/CourseQueries.hs

```diff
@@ -271,3 +273,22 @@ queryGraphs =
         do graphs :: [Entity Graph] <- selectList [] []
            return $ createJSONResponse graphs
 
+-- ========================================================
+
+getLectureTime :: [String] -> IO (Maybe [Time])
+getLectureTime [code, section, session] = runSqlite databasePath $ do 
```

Comment:
Make these three strings as separate arguments; lists should only be used if the input can truly be an arbitrary length.

Also, don't have a separate `runSqlite` for each of these, and `getTutorialTime`.  Instead, you want to just define the database action (`SqlPersistM`), and then surround the whole thing in a `runSqlite` where you use it in `getActiveTimetable`.